astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -646,6 +646,29 @@ def test_read_complex():
         Table.read(txt, format='ascii.ecsv')
 
 
+@pytest.mark.skipif(np.__version__.startswith('1.18.'),
+                    reason="numpy cannot parse 'complex' as string until 1.19+")
+def test_read_complex_v09():
+    """Test an ECSV file with a complex column"""
+    txt = """\
+# %ECSV 0.9
+# ---
+# datatype:
+# - {name: a, datatype: complex}
+# schema: astropy-2.0
+a
+1+1j
+2+2j"""
+
+    try:
+        Table.read(txt, format='ascii.ecsv')
+    except ValueError as ve:
+        if "datatype 'complex' of column 'a' is not in allowed values" in str(ve):
```

Comment:
You already skipping numpy above, what else would throw this error? Please consider rewriting the test.

Reply:
Any other error will fail, and should fail. The test should pass when the table is read successfully. 

This specific error comes from ECSV.py, not from numpy. As I write below, we are checking to make sure future changes don't break backwards compatibility. But the except blocks (both) might be unnecessary.

Reply:
Same as the other, it's removed, so resolving.